### PR TITLE
Change default branch from master to main

### PR DESCRIPTION
**Pull request by jsignell**
I'm realizing that doing distributed and dask at the same time is unfeasible since it'll mean that the tests never pass.  I am going to split this up into just the distributed part.

### CODE REVIEW

**Comment by jakirkham** [comment]
Just a note, we might want to add `main` to `pip install` commands that do development installs that are not otherwise specifying a branch. We needed to do this for `zict` recently as noted in PR ( https://github.com/dask/distributed/pull/4499 )

**Comment by jsignell** [comment]
Thanks @jakirkham. I just tried pip installing from git without specifying the branch and it seemed to work fine. 

```bash
conda create -n test pip python=3.7
conda activate test
pip install git+https://github.com/dask/zict.git
```

**Comment by jakirkham** [comment]
Interesting. Well here is the CI failure we saw

https://github.com/dask/distributed/runs/1868031469

**Comment by jakirkham** [comment]
Maybe a non-zero exit code is being set (even though it happens to work)?

**Comment by jsignell** [comment]
Definitely the windows part of #4499 replacing master with main needed to happen. I am just wondering if the travis part did.

**Comment by jakirkham** [comment]
Right was wondering if that happened when running locally

**Comment by jsignell** [comment]
I just rebased and I expect this to pass.

File: docs/release-procedure.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 Distributed follows a similar procedure for releasing as the core Dask project.
 
-See https://github.com/dask/dask/blob/master/docs/release-procedure.md for instructions.
\ No newline at end of file
+See https://github.com/dask/dask/blob/master/docs/release-procedure.md for instructions.
```
Comment (jrbourbeau) [inline docs/release-procedure.md:3]:
```suggestion
See https://github.com/dask/dask/blob/main/docs/release-procedure.md for instructions.
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jsignell. Given the recent release, and no reported issues with the branch name change in other `dask/*` repos, I think we're in a good place to merge this (and the corresponding PR in `dask/dask`) soon. I'll comment over in the community issue just to give people a heads up

File: docs/release-procedure.md (hunk 1 above)
Comment (jsignell) [reply docs/release-procedure.md:3]: I am purposefully only doing distributed in this PR. I wanted to be able to merge this one first, wait a bit and then do dask/dask.

Comment (jrbourbeau) [reply docs/release-procedure.md:3]: Gotcha, thanks for clarifying

**Comment by jsignell** [comment]
I think the failure is unrelated, so this should be good to merge.

**Review by jrbourbeau — APPROVED** [review]
Thanks @jsignell!

**Comment by jakirkham** [comment]
What's the ETA on fixing the Dask branch as well? We need to update some CI environments and would ideally like to make this change once

**Pull request merged**